src/mpid/ch4/netmod/ofi/ofi_recv.h

```diff
@@ -289,6 +289,58 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_do_irecv(void *buf,
 }
 
 
+#ifdef MPL_HAVE_GPU
+static inline int MPIDI_NM_do_gpu_recv(void *buf,
```

Comment:
This function actually does more then `gpu_recv`, it supports both cpu and gpu memory. For non-gpu memory, it just checks pointer property and call CPU transfer.

How about renaming it to `MPIDI_NM_enable_gpu_recv`, to make it more readable?